specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json

```diff
@@ -0,0 +1,591 @@
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "| Field                    | Supported operators  \n|---------------------|------------------------\n\r\n|requestSubmitTime | ge, le, eq, gt, lt\n"
+          },
+          {
+            "$ref": "#/parameters/SkipQueryParameter"
+          },
+          {
+            "$ref": "#/parameters/TopQueryParameter"
+          }
+        ],
+		
+		"x-ms-pageable": {
+			"nextLinkName": "nextLink"
+		},
+        "responses": {
+          "200": {
+            "description": "Ok. Sending Quota request status and details",
+            "schema": {
+              "$ref": "#/definitions/QuotaRequestDetailsByNameList"
+            }
+          },
+          "404": {
+            "description": "Not found. Quota requests for provided quotaRequest not found"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ExceptionResponse"
+            }
+          }
+		}
+      }
+    }
+  },
+  "definitions": {
+    "CreateGenericQuotaRequestParameter": {
+      "description": "Generic Quota change request information for a resource Provider.",
+      "properties": {
+        "limit": {
+          "description": "The requested limit.",
+          "type": "integer"
+        },
+        "name": {
+		  "description": "Name of the resource provide by the resource Provider. Please use this name property for quotaRequests.",
+          "properties": {
+            "localizedValue": {
+              "type": "string",
+              "description": "Resource display name.",
+              "readOnly": true
+            },
+            "value": {
+              "type": "string",
+              "description": "Resource name.",
+              "readOnly": true
+            }
+          }
+        },
+        "resourceType": {
+          "description": "The resource type for the specific resource provider. For example, BatchAI has a resource type of ‘dedicated lowPriority cluster’. Call the /usages API and use the resource type and resource name from that API. This is an optional field because some resources, like compute, do not have a resource type.",
+          "type": "string"
+        },
+        "unit": {
+          "description": "The units of the limit, such as - Count, Bytes, etc. Use the unit field provided in the Get quota response.",
+          "type": "string"
+        },
+        "quotaPeriod": {
+          "description": "The quota period over which the usage values are summarized, such as - P1D (Per one day), PT1M (Per one minute), PT1S (Per one second). This parameter is optional because, for some resources like compute, the period doesn’t matter.",
+          "type": "string"
+        }
+      }
+    },
+    "CreateGenericQuotaRequestParameters": {
```

Comment:
The way you modeled the paths implies this is a resource and should be modeled as such with id, name, type, properties...

Reply:
I will remove 4xx response code.

Reply:
The get request is - /subscriptions/{subscriptionId}/providers/Microsoft.Quota/providers/{providerId}/locations/{location}/usages
To get the current resource usage and limit (quota).
The Put is quotaRequests, which as same as you described - change the limit field to requested quota request. 
The name doesn't match, I agree.
AS quota is not a resource, it's an operation on resources provide by different RPs- such as - compute, batchAI, etc.
That's why, we followed the /usages pattern to get the resource information and used /quotaRequests for Put quotRequests an the Get /quotaRequests is just checking the status of the quota requests.
Please let me know, if it sounds fine?

Reply:
Updated the API to match Get /serviceLimits and Put /ServiceLimits